sigstore/sign.py

```diff
@@ -293,6 +310,43 @@ def sign_artifact(
             ),
         )
 
+        # Create the proposed hashedrekord entry
+        if self._signing_ctx._rekor.major_api_version == REKOR_V1_API_MAJOR_VERSION:
```

Comment:
What do you think about this:  RekorClient provides a `build_proposed_entry()` that takes artifatc_signature, b64_cert and hashed_input (and dsse/hashedrekord selection) as argument -- then this code (and the dsse code) could just call `self._signing_context._rekor.build_proposed_entry()`

Reply:
That's a good idea, especially since this patch is building the request body.

Reply:
Changed for just hashedrekord. dsse to come later.